tests/unit/phpDocumentor/UriTest.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor;
+
+final class UriTest extends \PHPUnit_Framework_TestCase
+{
+    public function testItShouldBuildAUri()
+    {
+        $uri = new Uri('file://foo');
+
+        $this->assertInstanceOf('phpDocumentor\Uri', $uri);
+        $this->assertObjectHasAttribute('uri', $uri);
+    }
+
+    public function testItShouldDiscardAnInvalidUri()
+    {
+        $this->setExpectedException('\InvalidArgumentException', 'Invalid uri');
```

Comment:
Did you know that you can use annotations in phpunit to set this?

```
/**
 * @expectedException InvalidArgumentException
 * @expectedExceptionMessage Invalid uri
 */
```

You don't have to change this but wanted to make you aware of this in case you didn't know :)